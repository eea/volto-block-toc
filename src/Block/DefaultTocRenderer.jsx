/**
 * View toc block.
 * @module components/manage/Blocks/ToC/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { map } from 'lodash';
import { List } from 'semantic-ui-react';
import { FormattedMessage, injectIntl } from 'react-intl';
import AnchorLink from 'react-anchor-link-smooth-scroll';

const RenderListItems = ({ items }) => {
  return map(items, (item) => {
    const { id, level, title } = item;
    return (
      item && (
        <List.Item key={id} className={`headline-${level}`}>
          <AnchorLink href={`#${id}`}>{title}</AnchorLink>
          {item.items?.length > 0 && (
            <List.List role="list">
              <RenderListItems items={item.items} />
            </List.List>
          )}
        </List.Item>
      )
    );
  });
};

/**
 * View toc block class.
 * @class View
 * @extends Component
 */
const View = ({ data, tocEntries }) => {
  return (
    <>
      {data.title && !data.hide_title ? (
        <h2>
          {data.title || (
            <FormattedMessage
              id="Table of Contents"
              defaultMessage="Table of Contents"
            />
          )}
        </h2>
      ) : (
        ''
      )}
      <List ordered={data.ordered} bulleted={!data.ordered}>
        <RenderListItems items={tocEntries} />
      </List>
    </>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  properties: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default injectIntl(View);
