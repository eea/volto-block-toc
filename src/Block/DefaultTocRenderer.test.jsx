import React from 'react';
import renderer from 'react-test-renderer';
import DefaultTocRenderer from './DefaultTocRenderer';

jest.mock('react-intl', () => ({
  injectIntl: jest.fn((Component) => Component),
}));

describe('DefaultTocRenderer', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <DefaultTocRenderer properties={{}} data={{ title: 'Heading title' }} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
    expect(component).toMatchSnapshot();
  });
});
