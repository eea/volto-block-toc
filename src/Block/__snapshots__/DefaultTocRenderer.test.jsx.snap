// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultTocRenderer renders correctly 1`] = `
Array [
  <h2>
    Heading title
  </h2>,
  <ul
    className="ui bulleted list"
    role="list"
  />,
]
`;

exports[`DefaultTocRenderer renders correctly 2`] = `
Array [
  <h2>
    Heading title
  </h2>,
  <ul
    className="ui bulleted list"
    role="list"
  />,
]
`;
